### PR TITLE
Refactor/200/tf scenarios updated1

### PR DESCRIPTION
**Pull request by JinLee794**
# Description

>Thank you for your contribution !

> Please include a summary of the change and which issue is fixed.
> Please also include the context.
> List any dependencies that are required for this change.

| Pipeline |
| - |
| |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

# Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: scenarios/secure-baseline-multitenant/terraform/hub/outputs.tf [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
   value = module.network.vnet_id
 }
 
-output "bastion_name" {
-  value = var.deployment_options.deploy_bastion ? module.bastion[0].name : null
+output "virtual_network" {
+  value = module.network.vnet
 }
 
+# output "bastion_name" {
```
Comment (ibersanoMS) [inline scenarios/secure-baseline-multitenant/terraform/hub/outputs.tf:17]: Remove this if it's no longer needed

File: scenarios/secure-baseline-multitenant/terraform/spoke/main.tf [hunk 2]
```diff
@@ -6,14 +6,16 @@ terraform {
...
-      version = ">=3.66.0"
+      version = "4.5.0"
     }
     azurecaf = {
       source  = "aztfmod/azurecaf"
       version = ">=1.2.23"
     }
   }
-  backend "azurerm" {}
+
```
Comment (ibersanoMS) [inline scenarios/secure-baseline-multitenant/terraform/spoke/main.tf:16]: Maybe just add a note that you should uncomment it if bringing your own hub

File: scenarios/secure-baseline-multitenant/terraform/main.tf [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  passthrough   = local.global_settings.passthrough
+  use_slug      = local.global_settings.use_slug
+}
+
+resource "azurerm_resource_group" "spoke" {
+  name     = azurecaf_name.caf_name_spoke_rg.result
+  location = var.location
+
+  tags = local.base_tags
+}
```
Comment (ibersanoMS) [inline scenarios/secure-baseline-multitenant/terraform/main.tf:54]: What's the reason this definition is here instead of in the spoke?

File: scenarios/secure-baseline-multitenant/terraform/spoke/network.tf
Comment (ibersanoMS) [inline scenarios/secure-baseline-multitenant/terraform/spoke/network.tf:1]: Is this essentially creating a "spoke-networking" resource group?

File: deployment/bicep/main.bicep
Comment (ibersanoMS) [inline deployment/bicep/main.bicep:1]: Why is this here?

File: deployment/bicep/parameters.json
Comment (ibersanoMS) [inline deployment/bicep/parameters.json:1]: Why is this here?

File: deployment/bicep/shared/azmon.bicep
Comment (ibersanoMS) [inline deployment/bicep/shared/azmon.bicep:1]: Why is this here?

File: deployment/bicep/shared/createvmwindows.bicep
Comment (ibersanoMS) [inline deployment/bicep/shared/createvmwindows.bicep:1]: Why is this here?

File: deployment/bicep/shared/shared.bicep
Comment (ibersanoMS) [inline deployment/bicep/shared/shared.bicep:1]: Why is this here?

File: deployment/bicep/shared/vm-nic.bicep
Comment (ibersanoMS) [inline deployment/bicep/shared/vm-nic.bicep:1]: Why is this here?

File: deployment/bicep/vnettest/vnetWithBastian.bicep
Comment (ibersanoMS) [inline deployment/bicep/vnettest/vnetWithBastian.bicep:1]: Why is this here?

File: deployment/bicep/vnettest/vnetWithOutBastian.bicep
Comment (ibersanoMS) [inline deployment/bicep/vnettest/vnetWithOutBastian.bicep:1]: Why is this here?

File: scenarios/secure-baseline-multitenant/terraform/hub/main.tf [hunk 4]
```diff
@@ -4,14 +4,16 @@ terraform {
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = ">=3.66.0"
+      version = "4.5.0"
```
Comment (ibersanoMS) [inline scenarios/secure-baseline-multitenant/terraform/hub/main.tf:7]: Why is this one 4.5.0 but the others are >=4.0?

File: scenarios/shared/terraform-modules/app-service/windows-web-app/module.tf
Comment (ibersanoMS) [inline scenarios/shared/terraform-modules/app-service/windows-web-app/module.tf:1]: Remove anything that's commented out and no longer being used

**Review by ibersanoMS — COMMENTED** [review]
Thanks so much for doing this refactoring! It looks great! Had a few questions and comments.

File: deployment/bicep/vnettest/vnetWithOutBastian.bicep
Comment (JinLee794) [reply deployment/bicep/vnettest/vnetWithOutBastian.bicep:1]: for all the bicep config in the `/deployment` directory, can we remove them? I believe these got included during the rebasing activities

File: scenarios/secure-baseline-multitenant/terraform/spoke/network.tf
Comment (JinLee794) [reply scenarios/secure-baseline-multitenant/terraform/spoke/network.tf:1]: its creating a spoke resource group. Moved the rg definition to the hub and spoke main.tf for better clarity

File: scenarios/secure-baseline-multitenant/terraform/hub/main.tf (hunk 4 above)
Comment (ibersanoMS) [reply scenarios/secure-baseline-multitenant/terraform/hub/main.tf]: @JinLee794 Does this have to be 4.5.0 or can it be >= 4.0.0?

**Review by ibersanoMS — APPROVED** [review]
